File: app/src/main/java/com/na/dgsw/gongyou_android/utils/EventBus.java
package com.na.dgsw.gongyou_android.utils;

import android.os.Handler;
import android.os.Looper;

import com.squareup.otto.Bus;

/**
 * Created by NA on 2020-07-07
 * skehdgur8591@naver.com
 */
public class EventBus extends Bus {
    private static EventBus instance;

    public static EventBus getInstance() {
        if (instance == null) {
            synchronized (EventBus.class) {
                if (instance == null) {
                    instance = new EventBus();
                }
            }
        }
        return instance;
    }

    private Handler handler = new Handler(Looper.getMainLooper());

    @Override
    public void post(final Object event) {
        if (Looper.myLooper() == Looper.getMainLooper()) {
            super.post(event);
        } else {
            handler.post(() -> EventBus.super.post(event));
        }
    }
}

File: app/src/main/java/com/na/dgsw/gongyou_android/utils/EventBus.kt
package com.na.dgsw.gongyou_android.utils

import android.os.Handler
import com.squareup.otto.Bus
import android.os.Looper

/**
 * Created by NA on 2020-07-07
 * skehdgur8591@naver.com
 */
class EventBus : Bus() {
    private val handler = Handler(Looper.getMainLooper())
    override fun post(event: Any) {
        if (Looper.myLooper() == Looper.getMainLooper()) {
            super.post(event)
        } else {
            handler.post { super@EventBus.post(event) }
        }
    }

    companion object {
        var instance: EventBus? = null
            get() {
                if (field == null) {
                    synchronized(EventBus::class.java) {
                        if (field == null) {
                            field = EventBus()
                        }
                    }
                }
                return field
            }
            private set
    }
}